rust/src/lib/unit_tests/ovn.rs

```diff
@@ -27,199 +26,184 @@ bridge-mappings: []
     let desired: OvnConfiguration = serde_yaml::from_str(
         r"---
-bridge-mappings:
-- localnet: net1
-  bridge: br1
-- localnet: net1
-  state: absent
-",
+        bridge-mappings:
+        - localnet: net1
+          bridge: br1
+        - localnet: net1
+          state: absent",
     )
     .unwrap();
 
     let current: OvnConfiguration = serde_yaml::from_str(
         r"---
-bridge-mappings:
-- localnet: net1
-  state: present
-  bridge: br1
-",
+        bridge-mappings:
+        - localnet: net1
+          bridge: br1",
     )
     .unwrap();
 
-    assert_eq!(
-        MergedOvnConfiguration::new(desired, current),
-        Err(NmstateError::new(
-            InvalidArgument,
-            "Duplicated `localnet` keys in the provided ovn.bridge-mappings"
-                .to_string()
-        ))
-    )
+    let result = MergedOvnConfiguration::new(desired, current);
+
+    assert!(result.is_err());
+
+    if let Err(e) = result {
+        assert_eq!(e.kind(), ErrorKind::InvalidArgument);
+    }
 }
 
 #[test]
 fn test_ovsdb_empty_string_to_ovn_bridge_mappings() {
     let input_string = "";
-    assert_eq!(
-        string_to_ovn_bridge_mappings(input_string.to_string()),
-        Ok(Vec::new())
-    )
+    let result = OvnBridgeMapping::try_from(input_string);
+    assert!(result.is_err());
+
+    if let Err(e) = result {
+        assert_eq!(e.kind(), ErrorKind::InvalidArgument);
+    }
 }
 
 #[test]
 fn test_ovsdb_string_without_localnet_to_ovn_bridge_mappings() {
     let input_string = ":br1";
-    assert_eq!(
-        string_to_ovn_bridge_mappings(input_string.to_string()),
-        Err(NmstateError::new(
-            InvalidArgument,
-            "expected `<localnet>:<bridge>`, got: :br1".to_string()
-        ))
-    )
+    let result = OvnConfiguration::try_from(input_string);
+
+    assert!(result.is_err());
+
+    if let Err(e) = result {
+        assert_eq!(e.kind(), ErrorKind::InvalidArgument);
+    }
 }
 
 #[test]
 fn test_ovsdb_string_without_bridge_to_ovn_bridge_mappings() {
     let input_string = "net1:";
-    assert_eq!(
-        string_to_ovn_bridge_mappings(input_string.to_string()),
-        Err(NmstateError::new(
-            InvalidArgument,
-            "expected `<localnet>:<bridge>`, got: net1:".to_string()
-        ))
-    )
+    let result = OvnConfiguration::try_from(input_string);
+
+    assert!(result.is_err());
+
+    if let Err(e) = result {
+        assert_eq!(e.kind(), ErrorKind::InvalidArgument);
+    }
 }
 
 #[test]
 fn test_ovsdb_string_to_ovn_bridge_mappings() {
     let input_string = "net1:br1";
+
     assert_eq!(
-        string_to_ovn_bridge_mappings(input_string.to_string()),
-        Ok(vec!(OvnBridgeMapping {
+        OvnBridgeMapping::try_from(input_string).unwrap(),
+        OvnBridgeMapping {
             localnet: "net1".to_string(),
             bridge: Some("br1".to_string()),
-            state: Some(OvnBridgeMappingState::Present)
-        }))
+            state: None
```

Comment:
what about people currently using this which already have configurations like:
```yaml
- localnet: net1
  state: present
  bridge: br1
```
? 

Will this keep working ? Maybe we should leave an e2e test behind with this config, to assert we don't break existing users.

Reply:
New patch only deprecate the use of `state: present`, still support it.